### PR TITLE
Avoid dumping all information in large clusters

### PR DESCRIPTION
**Pull request by wojtek-t**
Inspired by recent GKE 2k-node failures.

It can take even 2 hours to get all those logs - see:
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gke-large-cluster/21/consoleText

And the amount of logs is completely ridiculous and non-managable in this case.

@zmerlynn - FYI

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -2264,9 +2264,19 @@ func DumpAllNamespaceInfo(c *client.Client, namespace string) {
...
+	// If cluster is large, then the following logs are basically useless, because:
+	// 1. it takes tens of minutes or hours to grab all of them
+	// 2. there are so many of them that working with them are mostly impossible
+	// So we dump them only if the cluster is relatively small.
+	maxNodesForDump := 20
+	if nodes, err := c.Nodes().List(api.ListOptions{}); err == nil {
+		if len(nodes.Items) <= maxNodesForDump {
+			dumpAllPodInfo(c)
+			dumpAllNodeInfo(c)
+		}
```
Comment (gmarek) [inline test/e2e/framework/util.go]: add an info that you're skipping dumping because the cluster is too large.

Comment (wojtek-t) [inline test/e2e/framework/util.go]: done

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by wojtek-t** [comment]
Tests passing - merging manually as "test fix".

**Pull request merged**